src/table/system.rs

```diff
@@ -73,22 +73,21 @@ impl<View: SystemTableView> SystemTable<View> {
 
 // These parts of the UEFI System Table interface may only be used until boot
 // services are exited and hardware control is handed over to the OS loader
-#[allow(clippy::mut_from_ref)]
 impl SystemTable<Boot> {
     /// Returns the standard input protocol.
-    pub fn stdin(&self) -> &mut text::Input {
+    pub fn stdin(&mut self) -> &mut text::Input {
         unsafe { &mut *self.table.stdin }
     }
 
     /// Returns the standard output protocol.
-    pub fn stdout(&self) -> &mut text::Output {
-        let stdout_ptr = self.table.stdout as *const _ as *mut _;
+    pub fn stdout(&mut self) -> &mut text::Output {
+        let stdout_ptr = self.table.stdout as *mut () as *mut _;
         unsafe { &mut *stdout_ptr }
```

Comment:
```suggestion
        unsafe { &mut *self.table.stdout.cast() }
```
This may be a style issue though.

Reply:
Thank you, I didn't know about the `cast()` method.